Update to LLVM 5.0.1

milljm opened the pull request.

Commit:
Update to LLVM 5.0.1

moosebuild commented:
Job [Documentation](https://www.moosebuild.org/job/150093/) on 3af5951 wanted to post the following:

View the site [here](http://mooseframework.org/docs/PRs/129/site)

This comment will be updated on new commits.

Commit:
Update MPICH to 3.2.1

Attempt to see if Clang documentation works on Linux machines
if we update MPICH.

While trying to update our VM image, I discovered several
packages we were listing in the DEB control file as required,
but they no longer are. Removing those requirements.

milljm commented:
Nope.
